### PR TITLE
Add cliphist to the xclipboard layer.

### PR DESCRIPTION
**Pull request by xuhdev**
This is similar to #11481 but has all features integrated in xclipboard.

### CODE REVIEW

File: layers/+tools/xclipboard/packages.el [hunk 1]
```diff
@@ -13,8 +13,20 @@
...
     (spacemacs-xclipboard :location local)
+    (cliphist :requires ivy
+              :toggle xclipboard-enable-cliphist)
     ))
 
+(defun xclipboard/init-cliphist ()
+  (use-package cliphist
+    :init (spacemacs/set-leader-keys
+            "xP" 'cliphist-paste-item
+            "xR" (defun spacemacs/xclipboard-cliphist-paste-item-rectangle ()
```
Comment (duianto) [inline layers/+tools/xclipboard/packages.el]: Create a new `xclipboard/funcs.el` file with this header:
```
;;; funcs.el --- xclipboard layer functions file for Spacemacs.
;;
;; Copyright (c) 2012-2018 Sylvain Benner & Contributors
;;
;; Author: <author>
;; URL: https://github.com/syl20bnr/spacemacs
;;
;; This file is not part of GNU Emacs.
;;
;;; License: GPLv3
```
Replace `<author>` in the header with your name.
The xclipboard layers author included their email in the `packages.el` header.
https://github.com/syl20bnr/spacemacs/blob/9637ceb389a3db0450cec3831a38d4ab417a5912/layers/%2Btools/xclipboard/packages.el#L5
But it's up to you if you want to include yours.


Then add the function definition from the `xR` key binding, to the new `funcs.el` file, and have `xR` call the `'spacemacs/xclipboard-cliphist-paste-item-rectangle` function.

Comment (xuhdev) [reply layers/+tools/xclipboard/packages.el]: Updated.

**Comment by xuhdev** [comment]
@duianto Changelog also updated now.

**Comment by duianto** [comment]
Thank you for contributing to Spacemacs.
The PR has been cherry-picked into the `develop` branch. It's safe to delete your branch.
Commit: https://github.com/syl20bnr/spacemacs/commit/cd24a73429fcce6ae23ed1557d0f2562ce533f2d

**Pull request closed without merging**